.github/workflows/compilation_on_sgx.yml

```diff
@@ -288,64 +203,90 @@ jobs:
           wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
           sudo apt update
           sudo apt install -y libsgx-launch libsgx-urts
-          source /opt/intel/sgxsdk/environment
-
-      - name: Build Sample [basic]
-        run: |
-          cd samples/basic
-          ./build.sh
-          ./run.sh
 
-      - name: Build Sample [file]
+      - name: Build iwasm for testing samples
         run: |
-          cd samples/file
           mkdir build && cd build
-          cmake ..
+          cmake .. ${{ matrix.iwasm_make_options_run_mode }} ${{ matrix.iwasm_make_options_feature }}
           cmake --build . --config Release --parallel 4
-          ./src/iwasm -f wasm-app/file.wasm -d .
+          cd ../enclave-sample
+          make
+        working-directory: product-mini/platforms/${{ matrix.platform }}
 
-      - name: Build Sample [multi-thread]
+      - name: Get LLVM libraries
+        if: matrix.iwasm_make_options_run_mode == '$AOT_BUILD_OPTIONS'
+        id: retrieve_llvm_libs
+        uses: actions/cache@v3
+        with:
+          path: |
+            ./core/deps/llvm/build/bin
+            ./core/deps/llvm/build/include
+            ./core/deps/llvm/build/lib
+            ./core/deps/llvm/build/libexec
+            ./core/deps/llvm/build/share
+          key: ${{ matrix.llvm_cache_key }}
+
```

Comment:
Just a reminder, add `fail-on-cache-miss: true` which is able to avoid L229 - L231.